pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -1,65 +1,135 @@
-                for k in _get_single_spec_df.mapping_dict
-                if k in cv_param_info
-            }
-        )
+        if entry_tag == (f"{element_tag_prefix}PeptideEvidence"):
+            stage = 0
+            # Back to 0, so no cvParam gets written
+        elif entry_tag == (f"{element_tag_prefix}ModificationParams"):
+            stage = 0
+        elif entry_tag == (f"{element_tag_prefix}PeptideEvidenceRef"):
+            continue
+
+        elif stage == 1:
+            sequence, modifications, peptide_lookup = _peptide_lookup(
+                entry, entry_tag, sequence, modifications, peptide_lookup
+            )
+        elif stage == 2:
+            modification_mass_map, mod_name, fixed_mods = _get_modifications(
+                entry, entry_tag, modification_mass_map, mod_name, fixed_mods
+            )
+        elif stage == 3:
+            spec_results, spec_ident_items, spec_records = _spec_records(
+                entry,
+                entry_tag,
+                spec_results,
+                spec_ident_items,
+                spec_records,
+                mapping_dict,
+            )
+
+        elif entry_tag == (f"{element_tag_prefix}DBSequence"):
+            stage = 1
+            # Short before peptide_lookup begins
+        elif entry_tag == (f"{element_tag_prefix}AdditionalSearchParams"):
+            stage = 2
+        elif entry_tag == (f"{element_tag_prefix}Inputs"):
+            stage = 3
+            # Short before spec_record begins
+
+        elif entry_tag == (f"{element_tag_prefix}AnalysisSoftware"):
+            version = "comet_" + "_".join(
+                re.findall(r"([/d]*\d+)", entry.attrib["version"])
+            )
+
+        entry.clear()
+        if entry_tag == (f"{element_tag_prefix}SpectrumIdentificationList"):
+            return (
+                version,
+                peptide_lookup,
+                spec_records,
+                modification_mass_map,
+                fixed_mods,
+            )
 
-        spec_records.append(psm_level_dict)
-    return pd.DataFrame(spec_records)
+
+def _peptide_lookup(entry, entry_tag, sequence, modifications, peptide_lookup):
+    if entry_tag == (f"{element_tag_prefix}PeptideSequence"):
+        sequence = {"sequence": entry.text}
+    elif entry_tag == (f"{element_tag_prefix}Modification"):
+        mod = {"monoisotopicMassDelta": entry.attrib["monoisotopicMassDelta"]}
+        mod.update({"location": entry.attrib["location"]})
+        modifications.append(mod)
+    elif entry_tag == (f"{element_tag_prefix}Peptide"):
+        if len(modifications) > 0:
+            peptide_lookup[entry.attrib["id"]] = {"modifications": modifications}
+        else:
+            peptide_lookup[entry.attrib["id"]] = {"modifications": ""}
+        peptide_lookup[entry.attrib["id"]].update(sequence)
+        modifications = []
+        sequence = ""
+    return sequence, modifications, peptide_lookup
```

Comment:
don't understand why you reset sequence to be an empty string - especially if sequence can also be a dict (line 85)